### DISCUSSION TITLE
Cannot get to work with Laravel Valet

### DISCUSSION THREAD

**Discussion by brenjt** [qruto/laravel-wave#5] [Q&A] [answered]
I configured my settings exactly how you describe it in your documentation but after about 3 page refreshes I get nothing but 504 Gateway Timeouts. The site is broken until I restart redis. Once its reset and I reload the page 3 times it goes back to 504 errors

My nginx config looks like this for the valet site

```

    location ~ [^/]\.php(/|$) {
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
        fastcgi_pass "unix:/Users/<user>/.config/valet/valet.sock";
        fastcgi_index "/Users/<user>/.composer/vendor/laravel/valet/server.php";
        include fastcgi_params;
        fastcgi_param SCRIPT_FILENAME "/Users/<user>/.composer/vendor/laravel/valet/server.php";
        fastcgi_param PATH_INFO $fastcgi_path_info;
    }

    location = /wave {
        rewrite ^/wave$ /Users/<user>/.composer/vendor/laravel/valet/server.php?$query_string break;
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
        fastcgi_pass "unix:/Users/<user>/.config/valet/valet.sock";
        fastcgi_index "/Users/<user>/.composer/vendor/laravel/valet/server.php";
        include fastcgi_params;
        fastcgi_param SCRIPT_FILENAME "/Users/<user>/.composer/vendor/laravel/valet/server.php";
        fastcgi_param PATH_INFO $fastcgi_path_info;
        fastcgi_read_timeout 2m;
    }

```

As soon as I commented out `window.Wave = new Wave();` everything went back to normal.
I am using predis with redis version 7.0.0

What could I be doing wrong?
Does this not work with Laravel Valet or is this a Redis problem?

**Comment by slavarazum** [comment]
Hi! If you need setup a persistent connection on the `local` environment with Valet, try to set `fastcgi_read_timeout 2m;` for `location ~ [^/]\.php(/|$)` and remove `location = /wave` block.
In addition, check out [Persistent Connection / Fighting with Timeouts](https://github.com/qruto/laravel-wave#persistent-connection--fighting-with-timeouts) section to ensure that **ping** event works correctly.

**Reply by brenjt** [reply]
Hmm. Changing the timeout and setting up pings made no difference. I got this error if I tried to reload the page after it successfully connects `[pool valet] server reached pm.max_children setting`. I don't understand how PHP-FPM works with NGINX enough to debug this. Its a shame cause your package seems awesome. I'll have to give this a third shot when I have more time.

**Reply by slavarazum** [reply]
1. try to update package to the latest version:

```php
composer update qruto/laravel-wave
```

It ships auto-ping connections for `local` environment.

2. remove `location = /wave` nginx configuration.
3. restart Valet and Redis
4. Try again to reload a page with Wave connection

P.S. Thank you for the kind words about the package 😌

**Comment by ezequidias** [comment]
here the same thing is happening! after reloading the page an average of 3 times it keeps trying to reconnect !! and it only works if I restart the application!

this a Redis problem?

I've done tests with both PHP-FPM and Laravel Octane!

Congratulations on the package! it's very promising!

**Comment by slavarazum** [comment, marked as answer]
Hey @ezequidias.
I really appreciate your kind words. Lets break out this issue 💪

However I also use Valet and everything works fine, looks like not active Redis subscribers doesn't shooting down after page reload.

Could you check if the problem disappears after running the ping command:

```bash
php artisan sse:ping
```

as a next step if the issue still persists, try to run:

```bash
# for Mac
brew services restart redis

# for Linux
sudo service redis-server restart
```

Let me know which results you will get 🤝

@brenjt Do you have the same issue?

**Reply by ezequidias** [reply]
Here it is working perfectly now!!
I had done the wrong configuration of laravel echo in my frontend application!!

Thank you for everything!

**Reply by slavarazum** [reply]
Great to hear! I hope it serves you well. Feel free to provide feedback if you require any assistance.
